src/SSA_stepper.jl

```diff
@@ -145,15 +146,22 @@ DiffEqBase.add_tstop!(integrator::SSAIntegrator,tstop) = integrator.tstop = tsto
 function DiffEqBase.step!(integrator::SSAIntegrator)
     integrator.tprev = integrator.t
 
+    end_time = integrator.sol.prob.tspan[2]
+    next_jump_time = integrator.t >= integrator.tstop ? end_time : integrator.tstop
+
     doaffect = false
     if !isempty(integrator.tstops) &&
         integrator.tstops_idx <= length(integrator.tstops) &&
-        integrator.tstops[integrator.tstops_idx] < integrator.tstop
+        integrator.tstops[integrator.tstops_idx] < next_jump_time
 
         integrator.t = integrator.tstops[integrator.tstops_idx]
         integrator.tstops_idx += 1
     else
-        integrator.t = integrator.tstop
+        integrator.t = next_jump_time
+        if integrator.t >= end_time
```

Comment:
this isn't correct. If you call `step!` while at `end_time` it should keep stepping. You should able to use `step!` to step indefinitely. Basically if you keep calling `step!` it should do the same thing as `solve!` until the integration is done. Then you can keep `step!`ing beyond that just making sure to hit tstops.

Reply:
I see! I added a comment that addresses this concern.